### PR TITLE
Iosxe/show license

### PR DESCRIPTION
**Pull request by itdependsnetworks**

### CODE REVIEW

File: src/genie/libs/parser/iosxe/show_license.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+# =================
+# Schema for:
+#  * 'show license'
+# =================
+
+class ShowLicenseSchema(MetaParser):
+    """Schema for show license."""
+
+    schema = {
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/show_license.py:26]: schema looks good

File: src/genie/libs/parser/iosxe/show_license.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        for line in out.splitlines():
+            line_strip = line.strip()
+            if not line_strip.startswith("Index"):
+                try:
+                    data_type, value = line_strip.split(":", 1)
+                    regex = regex_map.get(data_type)
+                except ValueError:
+                    continue
+            else:
+                match = p1.match(line_strip)
```
Comment (badalsoyantar) [inline src/genie/libs/parser/iosxe/show_license.py:102]: Please add matching line as comment

File: src/genie/libs/parser/iosxe/show_license.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+    cli_command = 'show license'
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        p1 = re.compile(r"Index\s+(?P<id>\d+)\s+Feature:\s+(?P<feature>\S+)", re.MULTILINE)
```
Comment (badalsoyantar) [inline src/genie/libs/parser/iosxe/show_license.py:61]: re.MULTILINE is not required as we will do out.splitlines()

Comment (badalsoyantar) [reply src/genie/libs/parser/iosxe/show_license.py:61]: Need to add sample output line for all regex patterns

File: src/genie/libs/parser/iosxe/show_license.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
+''' show_license.py
+
+IOS parsers for the following show commands:
```
Comment (LenzDong) [inline src/genie/libs/parser/iosxe/show_license.py:3]: typo `IOSXE`

File: src/genie/libs/parser/iosxe/show_license.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        #         Period left: Life time
+        p2 = re.compile(r"\s+(?P<period_left>(Life\s+time|Not\s+Activated))")
+        # 	        Period Used: 0  minute  0  second
+        p3 = re.compile(r"\s+(?P<period_minutes>\d+)\s+minute\s+(?P<period_seconds>\d+)\s+second")
+        #         License Type: Permanent
+        p4 = re.compile(r"\s+(?P<license_type>(Permanent|EvalRightToUse))")
+        #         License State: Active, In Use
+        p5 = re.compile(
+            r"\s+((?P<count_in_use>\d+)/(?P<count_violation>\d+)\s+\(In-use/Violation\)|(?P<count>\S+))",
+            re.MULTILINE
```
Comment (LenzDong) [inline src/genie/libs/parser/iosxe/show_license.py:72]: please remove this line `re.MULTILINE`

**Comment by itdependsnetworks** [comment]
Should be good to go

**Pull request merged**